### PR TITLE
Feature/bidding v1

### PR DESCRIPTION
**Pull request by Stektpotet**
Basic implementation of #7 [Bidding v1](https://github.com/hackerspace-ntnu/Prosjekt-Spill-H22/issues/7)

For now, I've disabled the timer running the auction loop.
Instead the control flow can be managed manually through the context menu calls:

1. `Test Start Auction`
2. `GoToNextAuctionRound` (Creates a BiddingRound from the current AuctionStage object)
3. `Place Random Bids` (Place bids on a random item per player - can trigger simulated bidding wars)
    - . Repeat as many times as you like (and inspect _EDITORONLY_ActiveBiddingRound for insight in what's happening)
4. `FinishAuctionRound` (Distributes the items and draws tokens from player inventories according to bids placed)
5. `Test Stop Auction`

### CODE REVIEW

**Comment by toberge** [comment]
image

\>:(

File: Assets/Scripts/Auction/AuctionDriver.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
+using CollectionExtensions;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class AuctionDriver : MonoBehaviour
+{
+    [SerializeField] private int numberOfItems;
```
Comment (Fueredoriku) [inline Assets/Scripts/Auction/AuctionDriver.cs:8]: Maybe format declarations as
`
[SerializeField]

private int numberOfItems
`
instead, not that we have formally decided on using this semantic, so in the end it's up to you to decide 🤠

**Review by Fueredoriku — CHANGES_REQUESTED** [review]
Good work! 😎👍

I have the follwing suggestions:

- Change CoroutineTimer as we discussed with the ability for players to yield auctions and to add seconds to timer when a new bid arrives (up to a treshold)
- Add devcomments to custom editor attributes + readme for explaining their use in Attribute folder
- Add endlines in relevant scripts

And the optional:

-  Format [SerializeField] above field declaration

These suggestions are mostly nitpicking, and the structure/implementation is very good. Implementation was tested and code has been read.

File: Assets/Editor/PropertyDrawers/LabelAsDrawer.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            Debug.LogWarningFormat(
+                "{0}(\"{1}\") doesn't support arrays ",
+                typeof(LabelAsAttribute).Name,
+                labelAsAttribute.Text
+            );
+            EditorGUI.PropertyField(position, property, label);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (Stektpotet) [reply Assets/Editor/PropertyDrawers/LabelAsDrawer.cs]: Fixed in [f59aa2e](https://github.com/hackerspace-ntnu/Prosjekt-Spill-H22/pull/20/commits/f59aa2e30680aebf241e0cb11dca2edd06e5f583)

File: Assets/Scripts/Auction/AuctionDriver.cs (hunk 1 above)
Comment (Stektpotet) [reply Assets/Scripts/Auction/AuctionDriver.cs]: resolved in [b68490b](https://github.com/hackerspace-ntnu/Prosjekt-Spill-H22/pull/20/commits/b68490b8af2b410452cf2b92baaa2d744269efce)

**Comment by Stektpotet** [comment]
Ready for re-review! 🥳

**Review by Fueredoriku — APPROVED** [review]
Great work! 😁👍

**Pull request merged**